include/cell/copy/copy_atom.hpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file copy_atom.hpp
+ * @brief This file contains lightweight wrappers for hardware-accelerated copy
+ *        instructions.
+ */
+#pragma once
+
+#include "types/layout.hpp"
+
+namespace tiledcuda::cell::copy::atom {
+
+namespace tl = tiledcuda::cell::tile_layout;
+
+template <typename Element>
+struct LoadMatBase {
+    using DType = Element;
+
+    using ThreadLayout = tile_layout::ColMajor<16, 2>;
+
+    static constexpr int kAccessInBits = 128;  // 128 bits
+    static constexpr int kElmentBits = sizeof(DType) * 8;
+    static constexpr int kNumPerAccess = kAccessInBits / kElmentBits;
+
+    /// @brief returns the lane row of the current thread within a warp.
+    //         For an example, in ldmatrix, threads in a warp are arranged as
+    //         follows (a 16 x 2 column-major):
+    //
+    //         |  | 0 |  1|
+    //         |--|---|---|
+    //         |0 | 0 | 16|
+    //         |1 | 2 | 17|
+    //         |2 | 4 | 18|
+    //         |  |...|...|
+    //         |15| 15| 31|
+    //
+    //         if threadIdx.x is 43, then its lane row is 8, lane col is 0.
+    DEVICE int lane_row_id() {
+        int lane_id = threadIdx.x % warpSize;
+        return lane_id % tl::num_rows<ThreadLayout>;
+    }
+
+    /// @brief returns the lane col of the current thread within a warp.
+    DEVICE int lane_col_id() {
+        int lane_id = threadIdx.x % warpSize;
+        return lane_id / tl::num_rows<ThreadLayout>;
+    }
+
+    /// @brief a thin wrapper for executing ldmatrix instruction to load a
+    ///        `16x16` tile to register.
+    template <const int kRows, const int kCols>
+    DEVICE void ldmatrix(const DType* src, DType* dst) {
+        uint32_t* reg = reinterpret_cast<uint32_t*>(dst);
+        uint32_t smem_addr =
+            static_cast<uint32_t>(__cvta_generic_to_shared(src));
+
+        for (int i = 0; i < kRows; ++i) {
```

Comment:
Why are the two nested for loops necessary in the `ldmatrix` implementation?

Reply:
You're right. These loops are incorrect. A stupid error.

Reply:
> Why are the two nested for loops necessary in the `ldmatrix` implementation?

Fixed.

Reply:
> Why are the two nested for loops necessary in the `ldmatrix` implementation?

The previous loops are to support the use of `ldmatrix` for loading data types other than half-precision tiles, but I didn't add unittest for it. After careful consideration and verification of the tensor core's `wmma` input type, I added a constraint specifying that `ldmatrix` should only be used for half-precision floating-point numbers.